test/Parse/omit_return.swift

```diff
@@ -0,0 +1,1408 @@
+        return "\(one) + \(two)"
+    }
+}
+
+var fv_implicitMultilineClosure: () -> String {
+    {
+        let one = "big a"
+        let two = "little a"
+        return "\(one) + \(two)"
+    }
+}
+
+var fv_implicitWrong: String {
+    17 // expected-error {{cannot convert return expression of type 'Int' to return type 'String'}}
+}
+
+var fv_explicitWrong: String {
+    return 17 // expected-error {{cannot convert return expression of type 'Int' to return type 'String'}}
+}
+
+var fv_implicitMulti: String {
+    print("uh oh")
+    "shucks howdy" // expected-warning {{string literal is unused}}
+}
+
+var fv_explicitMulti: String {
+    print("okay")
+    return "all right"
+}
+
+var fv_effectfulUsed: String {
+    logAndReturn("okay")
+}
+
+// Unused returns
+
+var fv_effectfulIgnored: () {
+    logAndReturn("okay")
+}
+
+var fv_effectfulIgnoredVoid: Void {
+    logAndReturn("okay")
+}
+
+var fv_effectfulIgnoredSwiftVoid: Swift.Void {
+    logAndReturn("okay")
+}
+
+// Stubs
+
+var fv_stubEmptyTuple: () {
+    fatalError()
+}
+
+var fv_stubVoid: Void {
+    fatalError()
+}
+
+var fv_stubSwiftVoid: Swift.Void {
+    fatalError()
+}
+
+var fv_stubMyVoidTypealias: MyOwnVoid {
+    fatalError()
+}
+
+var fv_stubImplicitReturnNever: Never {
+    fatalError()
+}
+
+var fv_stubExplicitReturnNever: Never {
+    return fatalError()
+}
+
+var fv_stubExplicitReturnNeverAsMyOwnNever: MyOwnNever {
+    return fatalError() // expected-error {{cannot convert return expression of type 'Never' to return type 'MyOwnNever'}}
+}
+
+var fv_stubExplicitReturnMyOwnNeverAsNever: Never {
+    return myOwnFatalError() // expected-error {{cannot convert return expression of type 'MyOwnNever' to return type 'Never'}}
+}
+
+var fv_stubImplicitReturnNeverAsMyOwnNever: MyOwnNever {
+    fatalError()
+}
+
+var fv_stubImplicitReturnMyOwnNeverAsNever: Never {
+    myOwnFatalError()
+}
+
+var fv_stubString: String {
+    fatalError()
+}
+
+// Forced Trying
+
+var fv_forceTryUnusedExplicit: () {
+    return try! failableLogAndReturn("oh") //expected-error {{unexpected non-void return value in void function}}
+}
+
+var fv_forceTryUnusedImplicit: () {
+    try! failableLogAndReturn("uh") 
+}
+
+var fv_forceTryExplicit: String {
+    return try! failableIdentity("shucks")
+}
+
+var fv_forceTryImplicit: String {
+    try! failableIdentity("howdy")
+}
+
+// Optional Trying
+
+var fv_optionalTryUnusedExplicit: () {
+    return try? failableLogAndReturn("uh")  //expected-error {{unexpected non-void return value in void function}}
+}
+
+var fv_optionalTryUnusedImplicit: () {
+    try? failableLogAndReturn("oh") //expected-warning {{result of 'try?' is unused}}
+}
+
+var fv_optionalTryExplicit: String? {
+    return try? failableIdentity("shucks")
+}
+
+var fv_optionalTryImplicit: String? {
+    try? failableIdentity("howdy")
+}
+
+
+
+// MARK: - Free Properties : Get
+
+
+
+var fvg_nop: () {
+    get {
+    }
+}
+
+var fvg_missing: String {
+    get {
+    }
```

Comment:
Shouldn't this produce *some* diagnostic? Or does that not happen until SIL diagnostics, because it's dataflow-dependent?

Reply:
Yup!  Compiling the code with the debug build of these changes produces the error you're expecting:

```
error: missing return in a function expected to return 'String'
    }
    ^

```